src/server/pps/server/api_server.go

```diff
@@ -1549,8 +1566,11 @@ func (a *apiServer) validatePipelineRequest(request *pps.CreatePipelineRequest)
 	if request.TFJob != nil {
 		return goerr.New("embedding TFJobs in pipelines is not supported yet")
 	}
-	if request.S3Out {
-		return goerr.New("s3 gateway services for PPS pipelines are not supported yet")
+	if request.S3Out && ((request.Service != nil) || (request.Spout != nil)) {
+		return goerr.New("s3 output is not supported in spouts or services")
+	}
+	if request.S3Out && request.EnableStats {
+		return goerr.New("stats are not supported for pipelines that output via Pachyderm's S3 gateway")
```

Comment:
how come?

Reply:
We have no way of knowing which output files correspond to a given datum (in particular, in the case where you have a cross input and one of the cross elements is a regular PFS input, so that you have more than one datum but the workers are writing to `s3://out` throughout processing)